scripts/utils/package.js

```diff
@@ -4,28 +4,79 @@ const {statSync, readdirSync} = require('fs');
 const {execSync} = require('./exec');
 
 /**
- * Get single packages full path
- * @param  {string} pkg specific package name or full path
- * @param  {string} packagesDir path where packages are stored
- * @returns {array} array of full path strings
+ * Determines if a given path is a directory
+ *
+ * @param {string} dirPath full directory path
+ * @returns {boolean}
  */
-function getPackage(pkg, packagesDir = 'packages/node_modules/@ciscospark') {
-  // check if this is a valid path
+function isDirectory(dirPath) {
   try {
-    if (statSync(pkg).isDirectory()) {
-      const fullpath = path.resolve(pkg);
-      statSync(path.resolve(fullpath, 'package.json'));
-      return fullpath;
+    if (statSync(dirPath).isDirectory()) {
+      return true;
     }
   }
   catch (err) {
-    if (err.code === 'ENOENT') {
-      return getPackage(path.resolve(packagesDir, pkg));
+    return false;
+  }
+  return false;
+}
+
+/**
+ * Checks if the given path is a directory containing a package.json file
+ *
+ * @param {string} packageDirectory
+ * @returns {boolean}
+ */
+function isPackageDirectory(packageDirectory) {
+  if (!isDirectory(packageDirectory)) {
+    return false;
+  }
+  try {
+    const fullpath = path.resolve(packageDirectory);
+    if (statSync(path.resolve(fullpath, 'package.json')).isFile()) {
+      return true;
     }
   }
+  catch (err) {
+    return false;
+  }
   return false;
 }
 
+/**
+ * Get single packages full path
+ * @param  {string} pkg specific package name or full path
+ * @returns {string|boolean} full path string or false if failed
```

Comment:
This works, but kind of nags at me because the return type can change. Can we return empty string so it will still be falsey?